tiledb/sm/fragment/fragment_metadata.cc

```diff
@@ -142,6 +142,87 @@ FragmentMetadata::FragmentMetadata(
+    shared_ptr<const ArraySchema> array_schema,
+    std::vector<uint64_t>&& file_sizes,
+    std::vector<uint64_t>&& file_var_sizes,
+    std::vector<uint64_t>&& file_validity_sizes,
+    URI&& fragment_uri,
+    bool has_timestamps,
+    bool has_delete_meta,
+    bool has_consolidated_footer,
+    uint64_t sparse_tile_num,
+    uint64_t tile_index_base,
+    tdb::pmr::vector<tdb::pmr::vector<uint64_t>>&& tile_offsets,
+    tdb::pmr::vector<tdb::pmr::vector<uint64_t>>&& tile_var_offsets,
+    tdb::pmr::vector<tdb::pmr::vector<uint64_t>>&& tile_var_sizes,
+    tdb::pmr::vector<tdb::pmr::vector<uint64_t>>&& tile_validity_offsets,
+    tdb::pmr::vector<tdb::pmr::vector<uint8_t>>&& tile_min_buffer,
+    tdb::pmr::vector<tdb::pmr::vector<char>>&& tile_min_var_buffer,
+    tdb::pmr::vector<tdb::pmr::vector<uint8_t>>&& tile_max_buffer,
+    tdb::pmr::vector<tdb::pmr::vector<char>>&& tile_max_var_buffer,
+    tdb::pmr::vector<tdb::pmr::vector<uint8_t>>&& tile_sums,
+    tdb::pmr::vector<tdb::pmr::vector<uint64_t>>&& tile_null_counts,
+    std::vector<std::vector<uint8_t>>&& fragment_mins,
+    std::vector<std::vector<uint8_t>>&& fragment_maxs,
+    std::vector<uint64_t>&& fragment_sums,
+    std::vector<uint64_t>&& fragment_null_counts,
+    uint32_t version,
+    std::pair<uint64_t, uint64_t> timestamp_range,
+    uint64_t last_tile_cell_num,
+    NDRange&& non_empty_domain,
+    RTree&& rtree,
+    FragmentMetadata::GenericTileOffsets&& generic_tile_offsets,
+    FragmentMetadata::LoadedMetadata&& loaded_metadata)
+    : resources_(resources)
+    , memory_tracker_(memory_tracker)
+    , array_schema_(array_schema)
+    , array_schema_name_(array_schema->name())
+    , dense_(array_schema_->dense())
+    , file_sizes_(std::move(file_sizes))
+    , file_var_sizes_(std::move(file_var_sizes))
+    , file_validity_sizes_(std::move(file_validity_sizes))
+    , footer_size_(0)
+    , footer_offset_(0)
+    , fragment_uri_(std::move(fragment_uri))
+    , has_consolidated_footer_(has_consolidated_footer)
+    , last_tile_cell_num_(last_tile_cell_num)
+    , has_timestamps_(has_timestamps)
+    , has_delete_meta_(has_delete_meta)
+    , sparse_tile_num_(sparse_tile_num)
+    , loaded_metadata_(std::move(loaded_metadata))
+    , meta_file_size_(0)
+    , tile_offsets_mtx_(tile_offsets.size())
+    , tile_var_offsets_mtx_(tile_var_offsets.size())
+    , rtree_(std::move(rtree))
+    , tile_index_base_(tile_index_base)
+    , tile_offsets_(std::move(tile_offsets))
+    , tile_var_offsets_(std::move(tile_var_offsets))
+    , tile_var_sizes_(std::move(tile_var_sizes))
+    , tile_validity_offsets_(std::move(tile_validity_offsets))
+    , tile_min_buffer_(std::move(tile_min_buffer))
+    , tile_min_var_buffer_(std::move(tile_min_var_buffer))
+    , tile_max_buffer_(std::move(tile_max_buffer))
+    , tile_max_var_buffer_(std::move(tile_max_var_buffer))
+    , tile_sums_(std::move(tile_sums))
+    , tile_null_counts_(std::move(tile_null_counts))
```

Comment:
CI fails in these lines because `tiledb::pmr::vector` is not move constructible. Manually changing the initializers to say `tile_offsets_(std::move(tile_offsets), memory_tracker_->get_resource(MemoryType::TILE_OFFSETS))` would fix the errors, but it is tedious and causes duplication.

I don't understand why `pmr::vector` is not move-constructible. If you move the vector, doesn't the allocator also move?